larreco/HitFinder/HitFinderTools/PeakFitterGaussian_tool.cc

```diff
@@ -270,13 +268,222 @@ namespace reco_tool {
         parIdx += 3;
       }
 
+      if (fRefit && chi2PerNDF > fRefitThreshold &&
+          chi2PerNDF < std::numeric_limits<double>::infinity()) {
+
```

Comment:
There is a substantial amount of code nested within the `if`-statement curly braces.  It would be much clearer if you had a dedicated function to call:

```C++
if (fRefit && chi2PerNDF > fRefitThreshold ...) {
  refit(...);
}
```

Reply:
I have moved the setting/getting of the fit parameters into functions now which reduces the amount of code to call. If you want, I an wrap it all in a single refit function but that would be somewhat cumbersome due to the large number of parameters that need passing.